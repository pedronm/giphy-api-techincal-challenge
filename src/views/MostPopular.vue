<template>
  <div class="most-popular">
    <gif-grid
      title="Most Popular"
      msg="Here are some of the most popular GIFs available on GIPHY right now!"
      contentType="mostPopular"
    />
  </div>
</template>

<script>
import GifGrid from "@/components/GifGrid.vue";

export default {
  name: "MostPopular",
  components: {
    GifGrid
  }
};
</script>
