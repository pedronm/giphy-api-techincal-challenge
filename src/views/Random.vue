<template>
  <div class="random">
    <img alt="Giphy Icon" src="../assets/giphy-branding/GIPHY Logo 75px.png" />
    <gif-grid title="Random"
        msg="Here's a random collection of GIFs from GIPHY!"
        contentType="random"/>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";
import store from '../store'
import GifGrid from '../components/GifGrid.vue'

@Component({
    components: {
        GifGrid
    }
})
export default class Random extends Vue {
    constructor() {
        super();
        store.dispatch('fetchRandom');
    }
};
</script>
