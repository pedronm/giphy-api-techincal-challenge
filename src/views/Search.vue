<template>
  <div class="search">
    
    <gif-grid title="Search"
        msg="Search for GIFs on GIPHY!"
        contentType="searchResults"/>
  </div>
</template>

<script>
import GifGrid from "@/components/GifGrid.vue";

export default {
  name: "Search",
  components: {
    GifGrid
  }
};
</script>
