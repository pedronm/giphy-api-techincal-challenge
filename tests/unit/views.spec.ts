import { shallowMount } from "@vue/test-utils";
import MostPopular from "@/views/MostPopular.vue";
import Random from "@/views/Random.vue";
import Search from "@/views/Search.vue";

describe("MostPopular.vue", () => {
    it("renders", () => {
        const wrapper = shallowMount(MostPopular, {});
        wrapper.html()
        expect(wrapper.html()).toMatchSnapshot();
    });
});

describe("Random.vue", () => {
    it("renders", () => {
        const wrapper = shallowMount(Random, {});
        wrapper.html()
        expect(wrapper.html()).toMatchSnapshot();
    });
});

describe("Search.vue", () => {
    it("renders", () => {
        const wrapper = shallowMount(Search, {});
        wrapper.html()
        expect(wrapper.html()).toMatchSnapshot();
    });
});
