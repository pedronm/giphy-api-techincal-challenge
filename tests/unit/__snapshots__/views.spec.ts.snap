// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostPopular.vue renders 1`] = `
<div class="most-popular">
  <gif-grid-stub title="Most Popular" msg="Here are some of the most popular GIFs available on GIPHY right now!" contenttype="mostPopular"></gif-grid-stub>
</div>
`;

exports[`Random.vue renders 1`] = `
<div class="random">
  <gif-grid-stub title="Random" msg="Here's a random collection of GIFs from GIPHY!" contenttype="random"></gif-grid-stub>
</div>
`;

exports[`Search.vue renders 1`] = `
<div class="search">
  <gif-grid-stub title="Search" msg="Search for GIFs on GIPHY!" contenttype="searchResults"></gif-grid-stub>
</div>
`;
